### ISSUE TITLE
Notify user via blue bar aside the project on project listing page (grid view)

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#2545]
Need to show blue bar aside the project row to notify the user about the activity in the project.

Ref - https://github.com/appirio-tech/connect-app/issues/2344

**Comment by RishiRajSahu** [comment]
fyi - @maxceem

**Comment by maxceem** [comment]
@RishiRajSahu 

1. Do I understand it correctly that we show this line when user has any kind of notification related to the project?
2. When do we hide this blue line for a project?
    - a) when all notifications are seen
    - b) when all notifications are read
    - c) when a user opened project? - for this case I think we don't have a backend support.

**Comment by RishiRajSahu** [comment]
1. Yes we need to show in case there is any activity happened for the project
2. if `a` and `c` are different ? do we have support for `a`

**Comment by maxceem** [comment]
2. We have support for `a`. I'm not sure if this feature is 100% reliable but it's there. Also it's supported by the DB:
      ![image](https://user-images.githubusercontent.com/146016/46010178-5840e000-c0f4-11e8-9b48-2578ceb03fbb.png)

The difference between `a` and `c`. That `a` is marking each of notifications as `seen` in some conditions like opening notifications dropdown. While in `c` I was talking about hiding blue line for project when we just open the project, but we don't have to `see` all the notifications.

**Comment by RishiRajSahu** [comment]
@maxceem okay got your point, also got the understanding `seen` and `read` -
1.**Read** - If after opening notification dropdown user `checks` a single notification or `mark all as read`
2.**Seen** - On click of a notification respective page is opened

Now the question arises is,
suppose if user has more than one updates on the project and on seeing the blue line as soon as clicks project, is redirected to dashboard (Current Scenario) So we have to choose one of the below approach as per me - 
1.We can have the support of marking all notifications as read for that project once the user opens dashboard
**OR**
2.We as have different pages **Dashboard**, **Scope**, **ProjectPlan** we can either show cue on the `secondary tool bar` in case of **Dashboard** and **Scope** page and on `Phase(s)` in case of **Project Plan** page

I am inclined towards **2** but it needs to be refined to not miss anything and keep it consistent.

**Comment by maxceem** [comment]
@RishiRajSahu yep, your right regarding `read` and `seen` work.

Regarding two approaches:
> 1.We can have the support of marking all notifications as read for that project once the user opens dashboard

Yes, looks like this is not what we want to achieve in many places. For example, in the https://github.com/appirio-tech/connect-app/issues/1985 we want to mark notifications as read  only when user reach the subject of the notification.

> 2.

I also think we should keep doing it this way. So as I understand it means that as long as we have any notifications `unread` in the project we keep blue line against the project on the project list. If this is correct we can handle this issue without waiting until the rest of issues related to notifications cues are handled.

> it needs to be refined to not miss anything and keep it consistent.

Note. So far we have the next small incosistency regarding `red cue` dot. In top toolbar on the notifications dropdown bell we show it if we have some notifications `unread` AND we didn't open dropdown yet (or during some time).
While what we do for Phase `Discussions` tab, we show `red cue` dot as long there are `unread` notifications without any other conditions.

**Comment by maxceem** [comment]
@RishiRajSahu to sum up. I propose to show this blue line, as long particular project has any `unread` notifications. As for now logic to [mark notifications as read](https://github.com/appirio-tech/connect-app/issues/1985) is already implemented and merged to `dev` this should give pretty smooth experience to users.

What you think?

**Comment by RishiRajSahu** [comment]
@maxceem as per the latest changes in dev, we have correct logic to mark notifications as read on respective action done user, so yes I think a single check of having unread notification for a project should suffice the requirement of showing the blue bar.

**Comment by maxceem** [comment]
To sum up.

We have to show the blue bar aside project in the project list (grid view):

![](https://user-images.githubusercontent.com/8484061/44731583-5b669300-aa98-11e8-923b-be9ed6e87183.png)

In case the user have any unread notifications for that project.

We have all notification loaded in the Redux store and we can filter them by `projectId`

![image](https://user-images.githubusercontent.com/146016/51294876-e3513780-1a4f-11e9-9107-592d88a9c3e2.png)

**Comment by RishiRajSahu** [comment]
@maxceem yep we are very much clear with when to show blue bar, only thing to mention is when to mark notifications as read, so as I mentioned here https://github.com/appirio-tech/connect-app/issues/2545#issuecomment-424616608 we should mark them as read as per the target page/phase.
fyi - @vikasrohit

**Comment by maxceem** [comment]
@RishiRajSahu yeah, I think marking notifications as `read` is already handled by https://github.com/appirio-tech/connect-app/issues/1985. The logic is: we mark notifications as `read` when we the subject of notification is visible to a user.

**Comment by gets0ul** [comment]
Can I work on this issue?
Thanks

**Comment by maxceem** [comment]
Sure, it's on you.

**Comment by gets0ul** [comment]
PR is up and ready for review.

**Comment by maxceem** [comment]
Thanks, please have at look at review https://github.com/appirio-tech/connect-app/pull/2761#pullrequestreview-194852524

**Comment by gets0ul** [comment]
New commit is added to the PR.
Please check.
